MetaMask/metamask-extension
# [e2e] Setup MV3 specific test folder and add 1 simple testcase introducing Service Worker re-start

## Explanation
There are some specific behaviors for the MV3 build that need to be covered by our e2e tests, related to stopping and re-starting service worker between user actions.

For that, this PR sets the initial base work and adds one testcase, so this can be continued by adding more testcases in the future. The work consists on:
- Add mv3 flag for running e2e tests
- Include mv3 flag on circle ci job for `e2e:chrome:mv3`
- Add specific folder for MV3 tests
- Add simple testcase to check dapp interaction after a service worker re-start. 

At the moment, this will fail for 2 reasons:
  - Keep the user logged PR is still not merged
  - Popup issue after terminating service worker. See here: https://github.com/MetaMask/metamask-extension/issues/16554

Fixes #16610

## Screenshots/Screencaps



### Before



### After

![image](https://user-images.githubusercontent.com/54408225/203086743-0db5799a-2a06-4da8-8833-8e82feba6dd3.png)


## Manual Testing Steps
Check [circle ci ](https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/31189/workflows/abb51914-1393-4eb7-b0cd-aa14c533e1d3/jobs/814648/parallel-runs/6?filterBy=ALL)and see that the new test is run on the `e2e:chrome:mv3` job.

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
